### PR TITLE
[media][ios] remove `assets-library://` usage in favour of `ph://`

### LINKED ISSUE #12903: Support returning ph:// from MediaLibrary APIs
### Summary

On iOS Assets Library API is depreacted: https://developer.apple.com/documentation/assetslibrary. The MediaLibrary still uses the old `assets-library://` schema. It is incompatible with a few APIs like the sharing API and the FaceDetection API.

Found this comment here that alerted me to that: https://github.com/react-native-cameraroll/react-native-cameraroll/blob/master/ios/RNCAssetsLibraryRequestHandler.m#L68.

After a [Twitter conversation ](https://twitter.com/notbrent/status/1392898252513579010?s=20)we found these lines responsible: https://github.com/expo/expo/blob/fffbdb3161449764087dfa6d555ad1f88eb86dfe/packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m#L920-L926

I was quite confused about it as you can see here: https://stackoverflow.com/questions/67422493/getting-a-filesystem-uri-from-assets-in-expo-rn. 

The only workaround I found from the JS API was to copy assets to a cache directory. This is less than ideal as it slows things down quite a bit.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

41.0

### Environment

N/A

### Reproducible demo or steps to reproduce from a blank project

N/A

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

The `assets-library://` uri scheme has been deprecated since iOS 9 - our MediaLibrary implementation mostly uses PhotoKit which prefixes assets with a `ph://` scheme that utilizes part of the `asset.localIdentifier` to locate resources

The `assets-library` scheme appears in some logic in other modules as well, like FileSystem - ideally we could deprecate / add warnings for users who still use these schemes, although I would think it is a fairly safe change if users are relying on only our APIs

# How

Assets are formatted via the `_exportAsset` method before being returned to JS - updating this method and the `_assetUriForLocalId` method results in the correct uris being returned.

# Test Plan

- Created a sandbox environment and loaded up image assets into a list using `MediaLibrary.getAssetsAsync()`
- Ensured the `uri` returned from this method were prefixed with `assets-library://`
- updated the implementation to return a `ph://` uri scheme for a given asset
- Ensured that the `uri` field was updated with the new scheme, and that the images were still loading

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-media-library/CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 ### 🎉 New features
 
-- Added `isAvailableAsync` method ([#13418](https://github.com/expo/expo/pull/13418) by [@danielmark0116](https://github.com/danielmark0116))
+- Added `isAvailableAsync` method ([#13418](https://github.com/expo/expo/pull/13418) by [@danielmark0116](https://github.com/danielmark0116))˝
```
Comment (barthap) [inline packages/expo-media-library/CHANGELOG.md:9]: Must be a typo 😅 
```suggestion
- Added `isAvailableAsync` method ([#13418](https://github.com/expo/expo/pull/13418) by [@danielmark0116](https://github.com/danielmark0116))
```

**Review by barthap — APPROVED** [review]
Looks good to me 👍

**Review by tsapeta — APPROVED** [review]

**Review by lukmccall — APPROVED** [review]
LGMT 🚀

**Comment by giautm** [comment]
This PR is a breaking changes when upload files from MediaLibrary. It caused the error on iOS:

```
No suitable URL request handler found for ph-upload://ED7AC36B-A150-4C38-BB8C-B6D696F4F2ED

-[ABI43_0_0RCTNetworking networkTaskWithRequest:completionBlock:]
    ABI43_0_0RCTNetworking.mm:658
-[ABI43_0_0RCTNetworking processDataForHTTPQuery:callback:]
-[ABI43_0_0RCTHTTPFormDataHelper process:callback:]
-[ABI43_0_0RCTNetworking processDataForHTTPQuery:callback:]
-[ABI43_0_0RCTNetworking buildRequest:completionBlock:]
-[ABI43_0_0RCTNetworking sendRequest:callback:]
```

**Comment by barthap** [comment]
Hey @giautm !

Could you please open a new issue with minimal example on Snack, that works with SDK 42, but stopped working in SDK 43? If this PR is really breaking, that will help us to fix the issue correctly.\

**Comment by giautm** [comment]
I'm migrade from SDKv41. So, I think it still break on SDKv32. I workaround by JS code to restore the old behavior, I knowns it should call getAssetsInfoAsync() instead - but small change for quick fix.

**Pull request merged**